caiman/gui/gui_pyqtgraph_layout.py

```diff
@@ -508,6 +548,75 @@ def remove_single():
     
 pars_action.param('REMOVE SINGLE').sigActivated.connect(remove_single)
 
+def add_new_neuron():
+    global mode, mov, mov_residual, estimates
+    mode = 'add_neuron'
+    p2.setTitle("mode: %s" % (mode))
+    compute_residual_mov()    
+
+pars_action.param('ADD NEW NEURON').sigActivated.connect(add_new_neuron)
+
+def add_neuron_clicked():
+    global x,y,i,j,thrshcomp_line, new_components, mov, mov_residual
+    W_, H, mask, weight, new_components = nmf_extract()    
+    
+    # Visualize
+    weight = make_color_img(weight)    
+    contour = cv2.findContours(cv2.threshold(cv2.cvtColor(weight, cv2.COLOR_BGR2GRAY), np.int(thrshcomp_line.value()), 255, 0)[1], cv2.RETR_TREE,
+                                     cv2.CHAIN_APPROX_SIMPLE)[0]
+    image_temp = estimates.background_image.copy()
+    cv2.drawContours(image_temp, contour, -1, (128, 0, 128), 1)
+    cv2.circle(image_temp,(y,x),1,(255,0,0),-1)
+    img.setImage(image_temp, autoLevels=False) 
+    p2.plot(H[0], clear=True)
+    
+def nmf_extract(kernel_size=12, maxthr=0.25):
+    global x, y, mov, mov_residual
+    
+    # The original point
+    seed = np.zeros(shape=mov.shape[1:], dtype=np.uint8)
+    seed[x,y] = 1    
+    
+    # Apply dilation to get ROI
+    kernel = np.ones((kernel_size,kernel_size), np.uint8)
+    mask = cv2.dilate(seed, kernel)
+    W = np.reshape(mask,-1, order='C').copy()    
+    
+    # Apply NMF
+    pixels = (W==1)   
+    Yr = np.asarray((mov_residual-np.min(mov_residual, 0)).to_2D(order='C').T[pixels, :], dtype=np.float)
```

Comment:
You should create `Yr` directly only for the needed pixels instead of constructing the full residual movie and then extracting the `pixels` that are needed. The way it is right now it will be very memory intensive.